### DISCUSSION TITLE
Extending the Markdown parser for a new Vue component

### DISCUSSION THREAD

**Discussion by 6r1d** [vuejs/vitepress#1119] [Q&A] [answered]
Hello.

Currently, adding a new Vue component into a theme is [easy](https://vitepress.vuejs.org/guide/using-vue.html#registering-global-components-in-the-theme). In addition to that, I'd like to know if it is possible to add a [`markdown-it`](https://github.com/markdown-it/markdown-it) [syntax extension](https://markdown-it.github.io/markdown-it/#MarkdownIt.use) similar to the blocks displayed [here](https://vitepress.vuejs.org/guide/markdown.html#default-title), associating it with a new Vue component in VitePress theme.

The [container code](https://github.com/markdown-it/markdown-it-container/blob/master/dist/markdown-it-container.js) is helpful, but I don't understand how the `:::` syntax is defined.

I'd like to use said component for source code blocks, separated by the version, so writing a component name in Markdown each time would be inconvenient.

**Comment by brc-dd** [comment, marked as answer]
What kind of syntax do you want, can you show us an example. Is it something like this:

```md
::: foo bar
baz
:::
```

or simply `::: foo` without any props? 

If it's first one, then you can have a look at this example: https://github.com/vuejs/vitepress/issues/313#issuecomment-1146832422 or https://github.com/markdown-it/markdown-it-container#example or https://github.com/vuejs/vitepress/blob/main/src/node/markdown/plugins/containers.ts.

If it's second one, then you can simply do:

```ts
    md.use(container, 'foo', {
      render: (tokens: Token[], idx: number) =>
        tokens[idx].nesting === 1 ? `<div class="foo">\n` : `</div>\n`
    })
```

**Reply by 6r1d** [reply]
The example from https://github.com/vuejs/vitepress/issues/313 looks great, thank you!

The other part of my question is on organizing the code: is it normal to insert Vue components like `<CMP title="test" />` and expect the `markdown-it` to parse the text as a Vue component definition, or should I somehow explain it to Markdown-it?

**Reply by brc-dd** [reply]
It should recognise them by default.
